spirv_msl.cpp

```diff
@@ -9165,9 +9165,12 @@ void CompilerMSL::emit_glsl_op(uint32_t result_type, uint32_t id, uint32_t eop,
 
 	case GLSLstd450Normalize:
 		// MSL does not support scalar versions here.
+		// MSL has no implementation for normalize in the fast:: namespace for half2 and half3
 		// Returns -1 or 1 for valid input, sign() does the job.
 		if (expression_type(args[0]).vecsize == 1)
 			emit_unary_func_op(result_type, id, args[0], "sign");
+		else if (expression_type(args[0]).vecsize <= 3 && (get<SPIRType>(args[0]).basetype == spirv_cross::SPIRType::Half) )
```

Comment:
Does `get<SPIRType>(args[0])` work here? Isn't `args[0]` a SPIR-V variable? I would expect to see `expression_type(args[0])` here instead.

And given all the references to `expression_type(args[0])`, can we pull that outside the `if-else` chain here, assign it to a variable (`auto& exp_type = expression_type(args[0];`), and use that variable in this chain (`exp_type.vecsize`, `exp_type.basetype`)?

Reply:
Makes a lot more sense. I updated it.

Reply:
Thanks. LGTM now.